src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -272,6 +274,8 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             std::shared_ptr<ThrottledFuncTrailing<Control::ScrollPositionChangedArgs>> updateScrollBar;
         };
 
+        bool _acrylicToggle;

```

Comment:
Why add another bool here, rather than leverage the existing `UseAcrylic` `RUNTIME_SETTING`?

Reply:
Initially I used the `UseAcrylic RUNTIME_SETTING `and it seemed to work, but after testing Mouse Scroll Wheel to change Opacity and Adjust Opacity to change opacity, I stumbled upon some problems with this method.

**Problems:**
![runtime_acrylic_breaks_adjustopac](https://github.com/microsoft/terminal/assets/15957528/fb36cf23-3c7b-4cb8-bf55-7879260aac51)
![runtime_acrylic_breaks_scrollwheel](https://github.com/microsoft/terminal/assets/15957528/f000a464-932a-4a80-966a-bbd78424c1fc)

**The actual Terminal Preview behaviour with UseAcrylic on in settings.json in these scenarios:**
![runtime_acrylic_correct_adjustopacity](https://github.com/microsoft/terminal/assets/15957528/a1912424-9661-4a57-8f8d-9d93dd5cd4e6)
![runtime_acrylic_correct_scrollwheel](https://github.com/microsoft/terminal/assets/15957528/6da920b5-6fba-43e9-b3aa-e3d60c3e2d16)

When the user uses Mouse Scroll Wheel to change Opacity and Adjust Opacity to change opacity through the command palette the `ControlCore::_setOpacity `function is triggered. This function manually turns off the` _runtimeUseAcrylic` when opacity becomes 100. 

When the user scroll to opacity 100 and then scroll down again, the runtime acrylic goes false at 100 and would stay false when you go back under 100. Also when adjusting the opacity through the command palette similar thing happens when going to set background opacity to 100% option it stays false when going to 25%, 50%, etc.

Basically we can’t use the` UseAcrylic RUNTIME_SETTING`, because it becomes False when the user turns off transparency(opacity becomes 100)  by using Mouse Scroll Wheel to change Opacity and Adjust Opacity to change opacity and we can’t use `UseAcrylic RUNTIME_SETTING` to turn `_runtimeUseAcrylic `back on since it’s false. So I figured we need to use an external boolean besides the `UseAcrylic () RUNTIME_SETTING`.

Using an external boolean solves these issues.

Reply:
How would you have solved this problem?

Reply:
Hmm, so I tried something in ffc78620e.

Instead of setting `_runtimeUseAcrylic` based on the opacity, when the opacity changes, just do the `opacity < 1.0` when we actually need it. 

That _seems_ to work... I'll play with it more but that's a start

Reply:
@zadjii-msft Jaswir took 🗡️

Reply:
@zadjii-msft Are you a DnD player?

Reply:
I am, but that was more of a reference to 
![image](https://github.com/microsoft/terminal/assets/18356694/3ca05658-5626-4a2a-a5d1-152dd2ac6ca9)

Reply:
I see, I didn't play much Zelda to be honest ..